control_panel_api/models.py

```diff
@@ -221,22 +221,17 @@ def aws_role_name(self):
         raise NotImplementedError
 
     def aws_create(self):
-        services.attach_bucket_access_to_role(
-            self.s3bucket.name,
-            self.has_readwrite_access(),
-            self.aws_role_name(),
-        )
+        self.aws_update()
```

Comment:
Reasons why I didn't get rid of the `AccessToS3Bucket.aws_create()` method were:
1) To keep this 1:1 mapping between DB object creation and corresponding method name for the side effect in AWS
2) To avoid changes in the `views` layer as this is an internal change
3) To not change too many things at once (as we can always iterate on this)

Said that, I could just update [AppS3BucketViewSet#perform_create()](https://github.com/ministryofjustice/analytics-platform-control-panel/blob/master/control_panel_api/views.py#L135) and [UserS3BucketViewSet#perform_create()](https://github.com/ministryofjustice/analytics-platform-control-panel/blob/master/control_panel_api/views.py#L161) to call `instance.aws_update()` on create, but:
1) `aws_update` and `aws_update` would do the same thing
2) Would that be less confusing?

Reply:
Basically, I opted for keeping calling `aws_create()` when a DB instance is created and `aws_update()` when a DB instance is update, even if these two methods are the same. I don't know if it makes more sense.